tests/HashMapProperties.hs

```diff
@@ -168,6 +169,25 @@ pAlterInsert k = M.alter (const $ Just 3) k `eq_` HM.alter (const $ Just 3) k
 pAlterDelete :: Key -> [(Key, Int)] -> Bool
 pAlterDelete k = M.alter (const Nothing) k `eq_` HM.alter (const Nothing) k
 
+#if MIN_VERSION_containers(0,5,8)
+-- If we have a recent containers version compare against its alterF.
+
+pAlterFAdjust :: Key -> [(Key, Int)] -> Bool
+pAlterFAdjust k =
+  fromJust . M.alterF (Just . fmap succ) k `eq_`
+  fromJust . HM.alterF (Just . fmap succ) k
+
+pAlterFInsert :: Key -> [(Key, Int)] -> Bool
+pAlterFInsert k =
+  fromJust . M.alterF (const . Just . Just $ 3) k `eq_`
+  fromJust . HM.alterF (const . Just . Just $ 3) k
+
+pAlterFDelete :: Key -> [(Key, Int)] -> Bool
```

Comment:
Can you add benchmarks to show the difference between the rewritten `Identity` and a non rewritten version?

Reply:
Not a bad idea. Need tests too, ideally.

Reply:
Actually, I think the rewrites are sufficiently obvious improvements that we shouldn't need to compare. Just need to watch for regressions.